src/qibo/ui/bloch.py

```diff
@@ -0,0 +1,296 @@
+
+    def _axis(self):
+        """Helper method to `_create_sphere` to construct the sphere's axis."""
+        theta = np.linspace(0, 2 * np.pi, 100)
+        z = np.zeros(100)
+        x = np.sin(theta)
+        y = np.cos(theta)
+        return x, y, z
+
+    def _parallel(self, z):
+        """Helper method to `_create_sphere` to construct the sphere's parallels."""
+        theta = np.linspace(0, 2 * np.pi, 100)
+        z = np.full(100, z)
+        r = np.sqrt(1 - z[0] ** 2)
+        x = r * np.cos(theta)
+        y = r * np.sin(theta)
+        return x, y, z
+
+    def _meridian(self, phi):
+        """Helper method to `_create_sphere` to construct the sphere's meridians."""
+        theta = np.linspace(0, 2 * np.pi, 100)
+        x = np.sin(theta) * np.cos(phi)
+        y = np.sin(theta) * np.sin(phi)
+        z = np.cos(theta)
+        return x, y, z
+
+    def _create_sphere(self):
+        "This function builds an empty Bloch sphere."
+
+        # Empty sphere
+        self.ax.plot_surface(*self._sphere_surface(), color="lavenderblush", alpha=0.2)
+
+        # Axis
+        x, y, z = self._axis()
+        combinations_axis = [(x, y, z), (z, x, y), (y, z, x)]
+
+        # Axis lines
+        line, zeros = np.linspace(-1, 1, 100), np.zeros(shape=(100))
+        combinations_axis_line = [
+            (line, zeros, zeros),
+            (zeros, line, zeros),
+            (zeros, zeros, line),
+        ]
+
+        # Meridian and Parallel
+        phi = [(n + 1) * np.pi / 3 for n in range(6)]
+        lat = (0.4, -0.4, 0.9, -0.9)
+        meridian = lambda x: self.ax.plot(*self._meridian(x), color="darkgrey")
+        parallel = lambda x: self.ax.plot(*self._parallel(x), color="darkgrey")
+
+        # Axis, Axis lines, Meridians, Parallels
+        with mpl.rc_context(self.STYLE):
+            [self.ax.plot(*combinations_axis[i], color="darkgrey") for i in range(3)]
+            [
+                self.ax.plot(*combinations_axis_line[i], color="darkgrey")
+                for i in range(3)
+            ]
+            [meridian(p) for p in phi]
+            [parallel(l) for l in lat]
+
+        # Text
+        with mpl.rc_context(self.STYLE_TEXT):
+            self.ax.text(1.2, 0, 0, "x", ha="center")
+            self.ax.text(0, 1.2, 0, "y", ha="center")
+            self.ax.text(0, 0, 1.2, r"$|0\rangle$", ha="center")
+            self.ax.text(0, 0, -1.3, r"$|1\rangle$", ha="center")
+
+    # -----States and Vectors-----
+    def _coordinates(self, state):
+        """This function determines the coordinates of a qubit in the sphere."""
+
+        def _paulis_expectation(state):
+            sigma_X = hamiltonians.SymbolicHamiltonian(X(0))
+            sigma_Y = hamiltonians.SymbolicHamiltonian(Y(0))
+            sigma_Z = hamiltonians.SymbolicHamiltonian(Z(0))
+
+            x = sigma_X.expectation(state)
+            y = sigma_Y.expectation(state)
+            z = sigma_Z.expectation(state)
+            return x, y, z
```

Comment:
```suggestion
        def _paulis_expectation(state):
            sigma_X = SymbolicHamiltonian(X(0))
            sigma_Y = SymbolicHamiltonian(Y(0))
            sigma_Z = SymbolicHamiltonian(Z(0))

            x = sigma_X.expectation(state)
            y = sigma_Y.expectation(state)
            z = sigma_Z.expectation(state)
            return x, y, z
```

Does this function need to be defined inside a method?